test/runtests.jl

```diff
@@ -184,3 +184,49 @@ end
     @test nnz(LDLT) == nnz(LDLT.L) + length(LDLT.d)
   end
 end
+
+@testset "positive_semidefinite" begin
+  A = [0.   0.   0.   0.   0.   0.   0.   0.   1.   0.
+       0.   0.   0.   0.   0.   0.   0.   0.   1.   0.
+       2.   4.   5.   -2   4.   1.   2.   2.   2.   0.
+       0.   0.   0.   0.   1.   9.   9.   1.   7.   1.
+       3.   0.   0.   0.   0.   0.   0.   0.   0.   0.
+       1.   3.   2.   1.   4.   3.   1.   0.   0.   7.
+       -3.  8.   0.   0.   0.   0.   -2.  0.   0.   1.
+       0.   0.   0.   5.   7.   9.   0.   2.   7.   1.
+       3.   2.   0.   0.   0.   0.   1.   3.   3.   2.
+       0.   0.   0.   0.  -3   -4    0.   0.   0.   0. ]
```

Comment:
Actually there may be some scaling issues because if I try to set `A[2, 9] = 2.0 ` the residuals become bigger, I'll try to understand why

Reply:
A probably becomes indefinite and the dynamic regularization kicks in. How do you measure the residual?

Reply:
I think it's because if `A[2, 9] = 2.0` , `b` is no longer in the Image of `M=A'A` so this may be normal.

Reply:
It's safer to set `b = A * ones(10)`.